### ISSUE TITLE
Game crash related to scroll of mirror image

### ISSUE THREAD

**Issue by maciel310** [00-Evan/shattered-pixel-dungeon#193]
This will probably be hard to reproduce, and as such I'm not certain that this was the actual cause of the crash, but just had the game force close on me. I was in Pixel Mart on level 11, and a Mirror Image clone was in there as well. I clicked on an item in the shop to buy it (I was not standing next to it), and as my character walked over to it the clone walked over and stepped right on the item I had clicked on. Once I got to the item, the game crashed.

I'm guessing it was confused by there now being something else in the slot different from what I had originally clicked on.

Here is what looks like the stack trace from logcat. Hopefully this is enough to track down the issue:

FATAL EXCEPTION: Thread-3
Process: com.shatteredpixel.shatteredpixeldungeon, PID: 8060
java.lang.NullPointerException: Attempt to invoke virtual method 'int com.shatteredpixel.shatteredpixeldungeon.actors.hero.Hero.tier()' on a null object reference
at com.shatteredpixel.shatteredpixeldungeon.actors.mobs.npcs.MirrorImage.duplicate(MirrorImage.java:72)
at com.shatteredpixel.shatteredpixeldungeon.items.scrolls.ScrollOfMirrorImage.spawnImages(ScrollOfMirrorImage.java:92)
at com.shatteredpixel.shatteredpixeldungeon.items.scrolls.ScrollOfMirrorImage$DelayedImageSpawner.act(ScrollOfMirrorImage.java:134)
at com.shatteredpixel.shatteredpixeldungeon.actors.Actor.process(Actor.java:223)
at com.shatteredpixel.shatteredpixeldungeon.scenes.GameScene$2.run(GameScene.java:472)

**Comment by maciel310** [comment]
After playing a little longer, actually it doesn't seem related to the clicking on an item. I'm guessing it might be a bug with the new Unstable Spellbook? (Which is what created the Mirror Image clones.) It was crashing every few steps for me, regardless of what I did during those steps. Eventually it stopped crashing, and it happened to spawn additional clones at the same turn an enemy appeared and I attacked it. Might've gotten stuck on a crash loop trying to spawn those clones, and something changed (the attack? the enemy? something off screen?) that got it to finally spawn successfully.

Or it is possible that it was a combination of both. The initial crash causing the spawning to fail after starting the game back up.
